File: _old/8_Sort_People/JavaScript/Main.js
var sortPeople = function (names, heights) {
  let tmp = []
  let newNames = []

  for (let i = 0; i < heights.length; i++) {
    let obj = {
      name: names[i],
      height: heights[i]
    }

    tmp.push(obj)
  }

  tmp.sort((a, b) => b.height - a.height)
  for (let i = 0; i < tmp.length; i++) {
    newNames.push(tmp[i].name)
  }

  return newNames
};

console.log(sortPeople(["Mary", "John", "Emma"], [180, 165, 170]));

File: _old/8_Sort_People/Golang/Main.go
package main

import (
	"fmt"
	"sort"
)

func sortPeople(names []string, heights []int) []string {
	type people struct {
		name   string
		height int
	}

	tmp := make([]people, 0, len(names))

	for i := 0; i < len(names); i++ {
		t := people{
			name:   names[i],
			height: heights[i],
		}
		tmp = append(tmp, t)
	}

	sort.Slice(tmp, func(i, j int) bool { return tmp[i].height > tmp[j].height })

	ans := make([]string, 0, len(names))
	for _, t := range tmp {
		ans = append(ans, t.name)
	}
	return ans
}

func main() {

	x := sortPeople([]string{"Mary", "John", "Emma"}, []int{180, 165, 170})
	fmt.Println(x)
}
